test_conformance/api/test_api_min_max.cpp

```diff
@@ -491,47 +493,40 @@ int test_min_max_mem_alloc_size(cl_device_id deviceID, cl_context context, cl_co
-    error = clGetDeviceInfo( deviceID, CL_DEVICE_MAX_MEM_ALLOC_SIZE, sizeof( maxAllocSize ), &maxAllocSize, NULL );
-    test_error( error, "Unable to get max mem alloc size from device" );
+    /* Get the max mem alloc size, limit the alloc to half of the available
+     * memory */
+    maxAllocSize = get_device_info_max_mem_alloc_size(
+        deviceID, MAX_DEVICE_MEMORY_SIZE_DIVISOR);
+    memSize = get_device_info_global_mem_size(deviceID,
+                                              MAX_DEVICE_MEMORY_SIZE_DIVISOR);
 
-    error = clGetDeviceInfo( deviceID, CL_DEVICE_GLOBAL_MEM_SIZE, sizeof( memSize ), &memSize, NULL );
-    test_error( error, "Unable to get global memory size from device" );
+    if (memSize < maxAllocSize)
+    {
+        log_info("Global memory size is less than max allocation size, using "
+                 "that.\n");
+        maxAllocSize = memSize;
+    }
 
     if (memSize > (cl_ulong)SIZE_MAX) {
       memSize = (cl_ulong)SIZE_MAX;
     }
 
-    if( maxAllocSize < requiredAllocSize)
+    if ((0 == gIsEmbedded && maxAllocSize < 128 * 1024 * 1024)
+        || maxAllocSize < 1 * 1024 * 1024)
     {
-        log_error( "ERROR: Reported max allocation size is less than required %lldMB! (%llu or %lluMB, from a total mem size of %lldMB)\n", (requiredAllocSize / 1024) / 1024, maxAllocSize, (maxAllocSize / 1024)/1024, (memSize / 1024)/1024 );
+        log_error("ERROR: Reported max allocation size is less than required");
         return -1;
     }
 
-    requiredAllocSize = ((memSize / 4) > (1024 * 1024 * 1024)) ? 1024 * 1024 * 1024 : memSize / 4;
-
-    if (gIsEmbedded)
-        requiredAllocSize = (requiredAllocSize < 1 * 1024 * 1024) ? 1 * 1024 * 1024 : requiredAllocSize;
-    else
-    requiredAllocSize = (requiredAllocSize < 128 * 1024 * 1024) ? 128 * 1024 * 1024 : requiredAllocSize;
-
-    if( maxAllocSize < requiredAllocSize )
-    {
-        log_error( "ERROR: Reported max allocation size is less than required of total memory! (%llu or %lluMB, from a total mem size of %lluMB)\n", maxAllocSize, (maxAllocSize / 1024)/1024, (requiredAllocSize / 1024)/1024 );
-        return -1;
-    }
-
-    log_info("Reported max allocation size of %lld bytes (%gMB) and global mem size of %lld bytes (%gMB).\n",
-             maxAllocSize, maxAllocSize/(1024.0*1024.0), requiredAllocSize, requiredAllocSize/(1024.0*1024.0));
-
-    if ( memSize < maxAllocSize ) {
-        log_info("Global memory size is less than max allocation size, using that.\n");
-        maxAllocSize = memSize;
-    }
+    log_info("Reported max allocation size of %lld bytes (%gMB) and global mem "
+             "size of %lld bytes (%gMB).\n",
+             maxAllocSize, maxAllocSize / (1024.0 * 1024.0), memSize,
+             memSize / (1024.0 * 1024.0));
 
     minSizeToTry = maxAllocSize/16;
-    while (maxAllocSize > (maxAllocSize/4)) {
-
+    currentSize = maxAllocSize;
+    while (currentSize >= maxAllocSize / PASSING_FRACTION)
```

Comment:
It took me a while to figure out the name "PASSING_FRACTION", but I'm having trouble thinking of anything better. Maybe something like MAX_REDUCTION_FACTOR?

Reply:
Change to MAX_REDUCTION_FACTOR